src/test/scala/com/snowplowanalytics/refererparser/CorruptedRefererQuerystringTest.scala

```diff
@@ -13,31 +13,25 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package com.snowplowanalytics.refererparser
 
-// Specs2
-import org.specs2.mutable.Specification
-
-// Cats
+import cats.Eval
 import cats.effect.IO
+import org.specs2.mutable.Specification
 
 class CorruptedRefererQuerystringTest extends Specification {
 
-  // Our data
-  val refererUri = "http://www.google.com/search?q=Psychic+Bazaar&sugexp=chrome,mod=3&sourceid=chrome&ie=UTF-8"
-  val expected   = Some(SearchReferer("Google", Some("Psychic Bazaar")))
-
-  val parser = Parser.create[IO](
-    getClass.getResource("/referers.json").getPath
-  ).unsafeRunSync() match {
-    case Right(p) => p
-    case Left(f) => throw f
-  }
+  val resource   = getClass.getResource("/referers.json").getPath
+  val ioParser   = CreateParser[IO].create(resource).unsafeRunSync().fold(throw _, identity)
```

Comment:
as `Either` is a bifunctor (the instance may come from `cats.implicits` though), maybe it's better to just `leftMap` in those cases?

Reply:
I think the advantage of `fold` here is that it always returns a `Parser` at compile time, even if an error might be thrown at run time. That allows us to later do `parser.parse(x)`. 

`leftMap` returns an `Either` and then we have to handle it before we can call `parser.parse(x)`.